File: src/Core/Logistics.Application.Tenant/Commands/Load/UpdateLoad/UpdateLoadCommand.cs
using Logistics.Shared.Enums;
using MediatR;

namespace Logistics.Application.Tenant.Commands;

public class UpdateLoadCommand : IRequest<ResponseResult>
{
    public required string Id { get; set; }
    public string? Name { get; set; }
    public string? OriginAddress { get; set; }
    public double? OriginAddressLat { get; set; }
    public double? OriginAddressLong { get; set; }
    public string? DestinationAddress { get; set; }
    public double? DestinationAddressLat { get; set; }
    public double? DestinationAddressLong { get; set; }
    public decimal? DeliveryCost { get; set; }
    public double? Distance { get; set; }
    public string? AssignedDispatcherId { get; set; }
    public string? AssignedTruckId { get; set; }
    public string? CustomerId { get; set; }
    public LoadStatus? Status { get; set; }
}

File: src/Client/Logistics.OfficeApp/src/app/core/api/models/load/update-load.command.ts
import {AddressDto} from "../address.dto";
import {LoadStatus, LoadType} from "./enums";

export interface UpdateLoadCommand {
  id: string;
  name?: string;
  loadType?: LoadType;
  originAddress: AddressDto;
  originAddressLat: number;
  originAddressLong: number;
  destinationAddress: AddressDto;
  destinationAddressLat: number;
  destinationAddressLong: number;
  deliveryCost: number;
  distance: number;
  assignedDispatcherId: string;
  assignedTruckId: string;
  customerId?: string;
  status?: LoadStatus;
}
